packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -27,64 +252,41 @@ export default function Indicator({ children }) {
 
     if (!buildId) {
       if (isOnPrettyUrl) {
-        buildId = latestBuild?.id
+        setBuildId(latestBuild?.id)
       } else {
         // Match UUID from preview build URL https://build-af44185e-b8e5-11eb-8529-0242ac130003.gtsb.io
         const buildIdMatch = host.match(/build-(.*?(?=\.))/)
-        buildId = buildIdMatch && buildIdMatch[1]
+        if (buildIdMatch) {
+          setBuildId(buildIdMatch[1])
+        }
       }
     }
 
-    const defaultBuildInfo = {
-      createdAt: currentBuild?.createdAt,
-      orgId: siteInfo?.orgId,
-      siteId: siteInfo?.siteId,
-      buildId,
+    setBuildInfo({
+      currentBuild,
+      latestBuild,
+      siteInfo,
       isOnPrettyUrl,
-      sitePrefix: siteInfo?.sitePrefix,
+    })
+
+    if (shouldPoll.current) {
+      setTimeout(pollData, POLLING_INTERVAL)
     }
+  }, [])
 
-    if (!trackedInitialLoad) {
+  useEffect(() => {
+    if (buildInfo?.siteInfo && !trackedInitialLoad) {
+      const { siteInfo } = buildInfo
       trackEvent({
         eventType: `PREVIEW_INDICATOR_LOADED`,
-        orgId: defaultBuildInfo.orgId,
-        siteId: defaultBuildInfo.siteId,
+        orgId: siteInfo?.orgId,
+        siteId: siteInfo?.siteId,
         buildId,
       })
 
-      trackedInitialLoad = true
-    }
-
-    if (currentBuild?.buildStatus === `BUILDING`) {
-      setBuildInfo({
-        status: `BUILDING`,
-        ...defaultBuildInfo,
-      })
-    } else if (currentBuild?.buildStatus === `ERROR`) {
-      setBuildInfo({
-        status: `ERROR`,
-        errorBuildId: currentBuild?.id,
-        ...defaultBuildInfo,
-      })
-    } else if (buildId === currentBuild?.id) {
-      setBuildInfo({
-        status: `UPTODATE`,
-        ...defaultBuildInfo,
-      })
-    } else if (
-      buildId !== latestBuild?.id &&
-      latestBuild?.buildStatus === `SUCCESS`
-    ) {
-      setBuildInfo({
-        status: `SUCCESS`,
-        ...defaultBuildInfo,
-      })
-    }
-
-    if (shouldPoll.current) {
-      setTimeout(pollData, POLLING_INTERVAL)
+      setTrackedInitialLoad(true)
```

Comment:
Are you sure you want to re-render a component with a setState? I think you want to move it to a ref so you don't trigger an unnecessary state change.
```suggestion
      setTrackedInitialLoad(true)
```